tmk_core/common/action_code.h

```diff
@@ -153,6 +155,12 @@ typedef union {
         uint8_t  op     :2;
         uint8_t  kind   :4;
     } layer_bitop;
+    struct action_layer_mods
+    {
+        uint8_t  mods   :8;
```

Comment:
Why 8 for mods?  

If you have that, then you don't actually need to strip the left/right info, as it's 4 bits for the mods, and a 5th for the right/left. 

Additionally, if we allowed 5 bits for layers, that would allow for 0-31.  So .... we could get the best of both, actually.

Reply:
Just needed to fill up the rest of the space (probably not necessary but whatever).

And yeah, we could do that... but it would mean changing the `LM()` define, which would probably break VIA.

Reply:
Oh, actually, the above is completely wrong. `QK_LAYER_MODS` takes up 8 bits already, so we only have the other 8 to play with for `LM()`. So, not possible at all :(